.github/workflows/build.yaml

```
@@ -51,3 +40,62 @@ jobs:
           cache: 'npm'
       - run: npm ci
       - run: npm run build
+      - uses: actions/upload-artifact@v2
+        with:
+          name: dist
+          path: dist
+  e2e:
+    runs-on: ubuntu-latest
+    needs:
+      - build
+      - lint
+      - test
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: "0"
+      - uses: actions/download-artifact@v2
+        with:
+          name: dist
+          path: dist
+      - uses: cypress-io/github-action@v2
+        with:
+          start: npm run http-server
+      - uses: actions/upload-artifact@v2
+        with:
+          name: e2e-videos
+          path: cypress/videos
+  docker:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    needs:
+      - e2e
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: "0"
+      - uses: actions/download-artifact@v2
+        with:
+          name: dist
+          path: dist
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```

Currently, this will cause GitHub to push `latest` with every commit push in any branch because of `on: [push]` at the top. This is undesired.
```suggestion
      - name: Build and push Docker image
        uses: docker/build-push-action@v2
        with:
          context: .
          push: false
```